src/gradata/hooks/implicit_feedback.py

```
@@ -0,0 +1,107 @@
+                    data={
+                        "signals": [s["type"] for s in signals],
+                        "snippets": [s["snippet"] for s in signals[:3]],
+                        "message_preview": message[:200],
+                    },
+                    ctx=ctx,
+                )
+            except Exception:
+                pass
+
+        signal_names = ", ".join(s["type"] for s in signals)
+        return {"result": f"IMPLICIT FEEDBACK: [{signal_names}]"}
```

Ruff (0.15.9)

[warning] Consider moving this statement to an `else` block

(TRY300)